apps/daimo-mobile/src/logic/daimoContacts.ts

```diff
@@ -10,64 +10,70 @@ import {
 import { daimoChainFromId } from "@daimo/contract";
 import { Address } from "viem";
 
-import { env } from "../logic/env";
+import { getCachedEAccount } from "./addr";
+import { env } from "./env";
+import { useSystemContactsSearch } from "./systemContacts";
 import { Account } from "../model/account";
-import { getCachedEAccount } from "../view/shared/addr";
 
-export interface AccountRecipient extends EAccount {
-  type: "account";
-  originalMatch?: string;
+interface BaseDaimoContact {
   lastSendTime?: number;
 }
 
-interface EmailRecipient {
+export interface EAccountContact extends EAccount, BaseDaimoContact {
+  type: "eAcc";
+  originalMatch?: string;
+}
+
+interface EmailContact extends BaseDaimoContact {
   type: "email";
   email: EmailAddress;
-  name?: string; // TODO: populate from contact book
-  lastSendTime?: number;
+  name?: string;
 }
 
-interface PhoneNumberRecipient {
+interface PhoneNumberContact extends BaseDaimoContact {
   type: "phoneNumber";
   phoneNumber: PhoneNumber;
   name?: string;
-  lastSendTime?: number;
 }
 
-export type Recipient =
-  | AccountRecipient
-  | EmailRecipient
-  | PhoneNumberRecipient;
-
-export type ExternalRecipient = EmailRecipient | PhoneNumberRecipient;
+// A DaimoContact is a "contact" of the user in the app.
+// Includes EAccounts and system contacts (email, phone number) with added
+// context about the contact based on user's own context. (lastSendTime, etc.)
+export type DaimoContact = EAccountContact | EmailContact | PhoneNumberContact;
 
-// todo: rename recipient to contact or friend
+// An ExternalRecipient is a contact that is not a EAccount. (i.e. not an
+// on-chain account)
+export type SystemContact = EmailContact | PhoneNumberContact;
```

Comment:
👍 Contact is a great abstraction

update comment ExternalRecipient > SystemContact (or maybe MsgContact... since we'll get these from people typing in eg a phone number directly into the search bar, not just from the system contacts)

Reply:
done